docs/contributing/set-up-ide.md

```diff
@@ -0,0 +1,33 @@
+### Configure project for IDE (optional)
+
+In this section you will configure the project in order to make use of 
+integrated development environment's (IDE) features like syntax highlighting and code completion.
+
+## 1. Generate `go.mod` file
+
+Unlike typical Go project, Moby does not follow the [SemVer](https://semver.org/) versioning,
+opting in for [CalVer](https://calver.org/) instead.
+This required Moby to use `vendor.mod` and corresponding `vendor.sum` file 
+instead of standard `go.mod` and `go.sum`.
+
+This means that the IDEs like [GoLand](https://www.jetbrains.com/go/), which require `go.mod` 
+to properly recognize the directory as a Go project,
+will not index dependencies inside the vendor directory.
+As a result, the code completion might be broken.
+
+In order to generate dummy `go.mod` file that will instruct IDE to treat the
+Moby project directory as a Go project, execute the `vendor.sh` script:
```

Comment:
We should probably not recommend this step; I was actually planning to open a PR to remove the dummy `go.mod` after vendoring is complete, as the `go.mod` causes running tests to fail (due to the `go.mod` being just a "stub").

This project is not (yet) a go module (for various reasons, one of those because of the use of CalVer tags for our binary releases, which go modules doesn't allow).

To configure Goland to work with the code, it's important to:

- check out the source repository in `GOPATH` mode, using `github.com/docker/docker` (which is the name of the packages), e.g. `~/go/src/github.com/docker/docker`


- _disable_ go modules in the Goland configuration for the project:

<img width="1200" alt="Screenshot 2022-04-04 at 19 50 31" src="https://user-images.githubusercontent.com/1804568/161602530-22282862-793d-4dc5-8132-359a6bbb11aa.png">

- And (to prevent the IDE from indexing modules it may find in GOPATH, to disable "Use GOPATH that's defined in system environment" and "index entire GOPATH"

<img width="1200" alt="Screenshot 2022-04-04 at 19 50 31" src="https://user-images.githubusercontent.com/1804568/161603139-67816426-3578-4a5c-8f8e-4942ebf5a28f.png">

Reply:
Thanks for the tips! I will modify the PR to include these steps instead.

As for removal of the `go.mod` file after vendoring - does it mean that this file is necessary only to run `go mod vendor` command in the `vendor.sh` script? If it is so, would it make sense to remove it in the script's `EXIT` trap, so that it is cleaned up also when some command fails?

Reply:
Correct, yes; while `go mod` allows an alternative module-file to be specified (`-modfile=vendor.mod`), it still requires the whole repository to be a module (so a `go.mod` to be present). We had some discussion wether or not the `go.mod` should be removed afterwards and there were _some_ use-cases where it could be useful to keep, I think it's causing more problems than solving them, so I think we should update the script to remove it (for now at least).

We're in the hairy situation currently that (as mentioned) we cannot (yet) move the repository to be a module; adding a `go.mod` to this repository will break all existing users of this repository (`+incompatible` versions are not allowed if the repository has a `go.mod`). So the `vendor.mod` is an "intermediate step" towards being a module, but quite a lot more is needed (we have a path towards that, but details need to be worked out).